src/modules/auth/hooks/useNonceSync.js

```diff
@@ -0,0 +1,55 @@
+import { useEffect, useState, useCallback } from 'react';
+import { useQueryClient } from '@tanstack/react-query';
+import { useCurrentApplication } from '@blockchainApplication/manage/hooks';
+import { useCurrentAccount, useAccounts } from '@account/hooks';
+import useSettings from '@settings/hooks/useSettings';
+import { AUTH } from 'src/const/queries';
+import { useAuthConfig } from './queries';
+
+// eslint-disable-next-line max-statements
+const useNonceSync = () => {
+  const queryClient = useQueryClient();
+  const [currentApplication] = useCurrentApplication();
+  const [currentAccount] = useCurrentAccount();
+  const { setNonceByAccount, getNonceByAccount } = useAccounts();
+  const currentAccountAddress = currentAccount.metadata.address;
+  const currentAccountNonce = getNonceByAccount(currentAccountAddress);
+  const { mainChainNetwork } = useSettings('mainChainNetwork');
+  const chainID = currentApplication.chainID;
+  const customConfig = {
+    params: {
+      address: currentAccountAddress,
+    },
+  };
+  const baseUrl = mainChainNetwork?.serviceUrl;
+  const config = useAuthConfig(customConfig);
+  const authData = queryClient.getQueryData([AUTH, chainID, config, baseUrl]);
+  const onChainNonce = authData?.data.nonce;
+  const [accountNonce, setAccountNonce] = useState(onChainNonce);
+
+  // Store nonce by address in accounts store
+  const handleLocalNonce = (currentNonce) => {
+    let localNonce = parseInt(currentAccountNonce || 0, 10);
+    if (localNonce < currentNonce) {
+      localNonce = currentNonce;
+    }
+    setNonceByAccount(currentAccountAddress, localNonce);
+
+    setAccountNonce(localNonce);
+  };
+
+  useEffect(() => {
+    handleLocalNonce(parseInt(onChainNonce, 10));
+  }, [onChainNonce]);
+
+  // Call incrementNonce after transaction signing
+  const incrementNonce = useCallback(() => {
+    let localNonce = currentAccountNonce;
+    localNonce += 1;
+    setNonceByAccount(currentAccountAddress, localNonce);
+  }, []);
+
+  return { accountNonce, onChainNonce, incrementNonce };
```

Comment:
Since nonce is string in API, couldnt this hook expose it as a string as well, no need to mix types except for when doing calculations

Reply:
Fixed